### PR TITLE
Add videos categories api and auth pages

### PR DESCRIPTION
**Pull request by Maianki**
### Files to be checked :
- [src/App.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
- [src/components/Card/VideoCard/VideoCard.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/components/Chips/Chips.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-421ffc999f38c3bb352904f36d1d0180d6de563f7f9b43a9f6e67f8c0abb97f3)
- [src/components/Form/Input/Input.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-83e933f060abefe7a5fa219653b3f5bb2528667a88ea384d765a99955a2303cd)
- [src/components/Form/Label/Label.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-bb1e23fa913d5281a7db6ea04d9b0eb0d970e9dcb6a44016bb68a7c54a85e0d1)
- [src/components/Navbar/Navbar.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-547f8328c15a9a29000b7ef4690ba4f1039d148d5c208e5a48d37a2acd21982e)
- [src/context/videos-and-category-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-e908d307e43003c21fc1b5e293786084d5a800e13416f4278a8c13486a83afa9)
- [src/hooks/useDocumentTitle.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-b482c3b9551a64077a0ddb79f1b40d75541a28dd14286225b6d3bc75eedded97)
- [src/hooks/useFilteredData.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-62a0adff77c5d7751c5e5c636e8a8e2e2ff9f6b28c8dd3808fbb8047dc73ff26)
- [src/pages/Login/Login.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-da005ba6520f0f42b7e1052a931cea46e6703aceea8989e37fa93718ac033ad1)
- [src/pages/Signup/Signup.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-a981778bc61ac908a780e1c6ab71acef92e39ae1fe892d7093ad25d12d3efb3e)
- [src/pages/VideoListing/VideoListing.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-c29d4d8e24e33d34873170b7e41d0eb888347168ab292bcbce5ecbe427815b9d)
- [src/reducers/videosAndCategoryReducer.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1#diff-fb82f5e05f84147d6fd0692cd0e58be4cfb4a08e9f7780e64560da0780f4d198)
- [src/utils/getTrimVideoTitle.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-videos-categories-API-and-auth-pages?expand=1)

#### Live Site Preview
[Live site Preview](https://shuttle-motion-video-library-git-add-videos-cate-7e6382-maianki.vercel.app/)

### CODE REVIEW

File: src/reducers/videosAndCategoryReducer.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const videosAndCategoryInitialState = {
+  videos: [],
+  categories: [],
+  filterByCategory: "All",
+};
+
+export const videosAndCategoryReducer = (state, action) => {
+  const { type, payload } = action;
```
Comment (foolhardy21) [inline src/reducers/videosAndCategoryReducer.js]: minor suggestion - you can move this de-structuring in the function argument (line 7).

File: src/pages/Signup/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
+import React, { useState } from "react";
+import { Input, Label, Navbar } from "components";
+import { useDocumentTitle } from "hooks";
+import { BiEye, BiEyeSlash } from "assets";
+import { Link } from "react-router-dom";
+import styles from "./signup.module.css";
+
+export function Signup() {
+  useDocumentTitle("Sign Up");
```
Comment (foolhardy21) [inline src/pages/Signup/Signup.jsx:9]: good use of custom hook.

File: src/hooks/useFilteredData.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * @returns filtered data based on filter applied by user
+ *
+ */
+
+export function useFilteredData() {
+  const {
+    videosAndCategory: { videos, filterByCategory },
+  } = useVideosAndCategories();
+
+  console.log(filterByCategory);
```
Comment (foolhardy21) [inline src/hooks/useFilteredData.js]: you can remove `console.log`

File: src/hooks/useFilteredData.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const filteredData = JSON.parse(JSON.stringify(videos));
+  const getFilteredByCategory = (videos, filterByCategory) => {
```
Comment (foolhardy21) [inline src/hooks/useFilteredData.js]: suggestion - you can convert this function to one line using ternary.

File: src/context/videos-and-category-context.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import axios from "axios";
```
Comment (foolhardy21) [inline src/context/videos-and-category-context.js:1]: to name this file, you have chosen `kebab-case` but the reducer and other files have `camelCase`. So, you can go with `camelCase` for all files.

**Review by foolhardy21 — APPROVED** [review]
code looks clean and compact. nice work.

**Review by abhisheksahani-nova — APPROVED** [review]
Subtle and clean , readable code , good to merge and look into vinay suggestions too

File: src/hooks/useFilteredData.js (hunk 3 above)
Comment (Maianki) [reply src/hooks/useFilteredData.js]: Done

File: src/hooks/useFilteredData.js (hunk 4 above)
Comment (Maianki) [reply src/hooks/useFilteredData.js]: Sure, Will do that.

File: src/context/videos-and-category-context.js (hunk 5 above)
Comment (Maianki) [reply src/context/videos-and-category-context.js:1]: Okay, will do that I use kebab-case for context and reducer as told by Tanay. So I will rename the reducer to kebab-case.

File: src/reducers/videosAndCategoryReducer.js (hunk 1 above)
Comment (Maianki) [reply src/reducers/videosAndCategoryReducer.js]: This makes it more readable so that's why I destructured it in separate line.

**Comment by Maianki** [comment]
Thanks, @foolhardy21 @abhisheksahani-nova  for reviewing.

**Pull request merged**